chain-signatures/node/src/protocol/signature.rs

```diff
@@ -36,81 +42,54 @@ pub struct SignRequest {
     pub time_added: Instant,
 }
 
-/// Type that preserves the insertion order of requests.
-#[derive(Default)]
-pub struct ParticipantRequests {
-    requests: VecDeque<SignRequest>,
-}
-
-impl ParticipantRequests {
-    fn insert(&mut self, request: SignRequest) {
-        self.requests.push_back(request);
-    }
-
-    pub fn len(&self) -> usize {
-        self.requests.len()
-    }
-
-    pub fn is_empty(&self) -> bool {
-        self.len() == 0
-    }
-
-    pub fn pop_front(&mut self) -> Option<SignRequest> {
-        self.requests.pop_front()
-    }
-}
-
-#[derive(Default)]
 pub struct SignQueue {
-    unorganized_requests: Vec<SignRequest>,
-    requests: HashMap<Participant, ParticipantRequests>,
+    me: Participant,
+    sign_rx: Arc<RwLock<mpsc::Receiver<SignRequest>>>,
+    requests: HashMap<Participant, VecDeque<SignRequest>>,
 }
 
 impl SignQueue {
-    pub fn new() -> Self {
-        Self::default()
+    pub fn channel() -> (mpsc::Sender<SignRequest>, mpsc::Receiver<SignRequest>) {
+        mpsc::channel(MAX_SIGN_REQUESTS)
```

Comment:
What would the behavior be if we got more than 1024 requests? Will they be dropped or wait to be added?

Reply:
yeah, they'll be dropped but we'll have a whole entire sweep about adding these configs to the contract when we have more demand for it